### DISCUSSION TITLE
How can i get the response header?

### DISCUSSION THREAD

**Discussion by zhigangfang** [Kong/kong#7312] [Ideas and feature requests]
### Summary

Hello ~~~
I'm develop go plugin, kong.Response.GetHeader can not work in access phase, and go-pdk do not support header_filter and body_filter phase. so, how can i get the response header before log phase?

### Additional Details

- Kong version v2.0.4
- go-pdk version v0.3.1

**Comment by jeremyjpj0916** [comment]
Correct, the access phase is only intended for manipulation of request data prior to the ```proxy_pass``` directive that routes Kong's http traffic. As for the go-pdk, Kong will likely be rolling out more functionality that will enable header and body filter phases, but Kong team will have to fill you in there on timelines or plans.

**Comment by bungle** [comment]
Yes, currently Go implementation is based on `cosockets` that are not unfortunately enabled on `header_filter` / `body_filter` (and `log` phase we work-around with `timer`, but that also causes some issues as missing the request context).
